File: analyze_inclusivity.js
const fs = require('fs');

// Read the SQL files
const mainFile = fs.readFileSync('ascevo/supabase/lessons_all_pillars.sql', 'utf8');
const fitnessFile = fs.readFileSync('.kiro/specs/lesson-content-enhancement/fitness_pillar.sql', 'utf8');

const allContent = mainFile + '\n' + fitnessFile;

// Extract character names (pattern: 'Name, Age,)
const namePattern = /'([A-Z][a-z]+), (\d{2}),/g;
const names = [];
let match;

while ((match = namePattern.exec(allContent)) !== null) {
  names.push({ name: match[1], age: match[2] });
}

// Count unique names
const uniqueNames = [...new Set(names.map(n => n.name))];

console.log('=== CHARACTER NAME DIVERSITY ANALYSIS ===\n');
console.log(`Total character examples found: ${names.length}`);
console.log(`Unique names: ${uniqueNames.length}\n`);

// Count frequency
const nameCounts = {};
names.forEach(n => {
  nameCounts[n.name] = (nameCounts[n.name] || 0) + 1;
});

console.log('Name frequency:');
Object.entries(nameCounts)
  .sort((a, b) => b[1] - a[1])
  .forEach(([name, count]) => {
    console.log(`  ${name}: ${count}`);
  });

console.log('\n=== BODY IMAGE & FITNESS LANGUAGE CHECK ===\n');

// Check for problematic body image language
const problematicPatterns = [
  /\b(fat|skinny|thin|obese)\b/gi,
  /\b(lose weight|weight loss|diet|dieting)\b/gi,
  /\b(beach body|summer body|bikini body)\b/gi,
  /\b(attractive|unattractive|ugly|beautiful body)\b/gi,
  /\b(perfect body|ideal body|body goals)\b/gi,
];

const fitnessContent = fitnessFile;
console.log('Checking Fitness pillar for body image language...\n');

problematicPatterns.forEach((pattern, idx) => {
  const matches = fitnessContent.match(pattern);
  if (matches) {
    console.log(`Pattern ${idx + 1} matches:`, matches);
  }
});

console.log('\n=== GENDER/SEXUALITY/ABILITY ASSUMPTIONS CHECK ===\n');

// Check for gendered assumptions
const genderedPatterns = [
  /\b(boyfriend|girlfriend|husband|wife)\b/gi,
  /\b(he should|she should|men should|women should)\b/gi,
  /\b(boys|girls) (are|do|don't)\b/gi,
];

console.log('Checking for gendered assumptions...\n');
genderedPatterns.forEach((pattern, idx) => {
  const matches = allContent.match(pattern);
  if (matches) {
    console.log(`Gendered pattern ${idx + 1}:`, matches.slice(0, 10));
  }
});

console.log('\n=== CONTEXT & BACKGROUND DIVERSITY CHECK ===\n');

// Check for diverse contexts
const contexts = [
  /\b(student|university|college|class|seminar)\b/gi,
  /\b(work|job|career|office|professional)\b/gi,
  /\b(family|parent|mother|father|sibling)\b/gi,
  /\b(friend|friendship|social)\b/gi,
];

contexts.forEach((pattern, idx) => {
  const matches = allContent.match(pattern);
  if (matches) {
    console.log(`Context ${idx + 1}: ${matches.length} references`);
  }
});

console.log('\n=== COMPLETED ===');

File: analyze_inclusivity.py
#!/usr/bin/env python3
"""
Analyze lesson content for inclusivity and sensitivity issues.
"""
import re
from collections import Counter
from pathlib import Path

def extract_names_from_sql(filepath):
    """Extract all character names and ages from SQL file."""
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Pattern: 'Name, age, ...
    pattern = r"'([A-Z][a-z]+), (\d{2}),"
    matches = re.findall(pattern, content)
    
    return [(name, int(age)) for name, age in matches]

def analyze_name_diversity(names):
    """Analyze diversity of names used."""
    name_list = [name for name, _ in names]
    name_counts = Counter(name_list)
    
    # Categorize names by origin (simplified)
    european_names = ['Lena', 'Marcus', 'Jake', 'Sofia', 'Aisha', 'Ryan', 'Priya', 'Ella', 
                      'Noah', 'Chloe', 'James', 'Mia', 'Ava', 'Leo', 'Grace', 'Daniel',
                      'Tom', 'Zara', 'Kai', 'Lily', 'Liam', 'Elena', 'Sophia', 'Mia',
                      'Anya', 'Luca', 'Ethan']
    
    south_asian = ['Aisha', 'Priya', 'Kai']
    arabic_middle_eastern = ['Aisha']
    east_asian = ['Kai']
    african = ['Zara']
    latin_american = ['Sofia', 'Elena', 'Lily', 'Luca']
    
    print(f"\n{'='*60}")
    print("NAME DIVERSITY ANALYSIS")
    print(f"{'='*60}")
    print(f"\nTotal unique names: {len(name_counts)}")
    print(f"Total name uses: {sum(name_counts.values())}")
    print(f"\nMost common names:")
    for name, count in name_counts.most_common(10):
        print(f"  {name}: {count} times")
    
    return name_counts

def search_body_image_language(filepath):
    """Search for potentially harmful body image language."""
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Problematic phrases to flag
    red_flags = [
        r'lose weight',
        r'get skinny',
        r'burn fat',
        r'beach body',
        r'summer body',
        r'six pack',
        r'shredded',
        r'ripped',
        r'toned',
        r'slim down',
        r'diet\b',
        r'calories.*restrict',
        r'cheat\s+(day|meal)',
        r'guilty.*eat',
        r'bad food',
        r'good food'
    ]
    
    findings = []
    for pattern in red_flags:
        matches = re.finditer(pattern, content, re.IGNORECASE)
        for match in matches:
            start = max(0, match.start() - 100)
            end = min(len(content), match.end() + 100)
            context = content[start:end].replace('\n', ' ')
            findings.append((pattern, context))
    
    return findings

def search_assumption_language(filepath):
    """Search for language that makes assumptions about gender, sexuality, ability."""
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Patterns that might indicate assumptions
    check_patterns = [
        r'his girlfriend|her boyfriend',
        r'his wife|her husband',
        r'normal\s+people',
        r'everyone can',
        r'just\s+\w+',  # "just run", "just lift" - minimizes difficulty
        r'simply\s+\w+',
    ]
    
    findings = []
    for pattern in check_patterns:
        matches = re.finditer(pattern, content, re.IGNORECASE)
        for match in matches:
            start = max(0, match.start() - 100)
            end = min(len(content), match.end() + 100)
            context = content[start:end].replace('\n', ' ')
            findings.append((pattern, context))
    
    return findings

def main():
    print("INCLUSIVITY & SENSITIVITY REVIEW")
    print("=" * 60)
    
    # Analyze main lessons file
    main_sql = Path("ascevo/supabase/lessons_all_pillars.sql")
    fitness_sql = Path(".kiro/specs/lesson-content-enhancement/fitness_pillar.sql")
    
    all_names = []
    
    if main_sql.exists():
        print(f"\nAnalyzing {main_sql}...")
        names = extract_names_from_sql(main_sql)
        all_names.extend(names)
        print(f"Found {len(names)} character examples")
    
    if fitness_sql.exists():
        print(f"\nAnalyzing {fitness_sql}...")
        names = extract_names_from_sql(fitness_sql)
        all_names.extend(names)
        print(f"Found {len(names)} character examples")
    
    # Analyze name diversity
    analyze_name_diversity(all_names)
    
    # Check age distribution
    ages = [age for _, age in all_names]
    print(f"\n{'='*60}")
    print("AGE DISTRIBUTION")
    print(f"{'='*60}")
    print(f"Age range: {min(ages)} - {max(ages)}")
    print(f"Average age: {sum(ages) / len(ages):.1f}")
    age_counts = Counter(ages)
    print(f"\nAge frequency:")
    for age in sorted(age_counts.keys()):
        print(f"  {age}: {'*' * age_counts[age]}")
    
    # Body image language check (Fitness pillar)
    if fitness_sql.exists():
        print(f"\n{'='*60}")
        print("BODY IMAGE LANGUAGE CHECK (Fitness Pillar)")
        print(f"{'='*60}")
        findings = search_body_image_language(fitness_sql)
        if findings:
            print(f"\n⚠️  Found {len(findings)} potential concerns:")
            for pattern, context in findings[:5]:  # Show first 5
                print(f"\nPattern: {pattern}")
                print(f"Context: ...{context}...")
        else:
            print("\n✓ No problematic body image language detected")
    
    # Assumption language check
    print(f"\n{'='*60}")
    print("ASSUMPTION LANGUAGE CHECK")
    print(f"{'='*60}")
    
    for filepath in [main_sql, fitness_sql]:
        if filepath.exists():
            findings = search_assumption_language(filepath)
            if findings:
                print(f"\n{filepath.name}:")
                print(f"  Found {len(findings)} patterns to review")
            else:
                print(f"\n{filepath.name}: ✓ No concerning patterns")

if __name__ == "__main__":
    main()
